internal/integration/infra_sdk_emitter.go

```diff
@@ -122,8 +118,30 @@ func (e *InfraSdkEmitter) emitSummary(i *integration.Integration, metric Metric,
 		ps.AddQuantile(*q.Quantile, *q.Value)
 	}
 
-	i.HostEntity.AddMetric(ps)
+	return e.addMetricToEntity(i, metric, ps)
+}
+
+func (e *InfraSdkEmitter) addMetricToEntity(i *integration.Integration, metric Metric, m metrics.Metric) error {
+	entityName, entityType, err := e.definitions.getEntity(metric)
+	// if we can't find an entity for the metric, add it to the "host" entity
+	if err != nil {
+		i.HostEntity.AddMetric(m)
+		return nil
+	}
+
+	// try to find the entity and add the metric to it
+	// if there's no entity with the same name yet, create it and add it to the integration
+	entity, ok := i.FindEntity(entityName)
+	if !ok {
+		entity, err = i.NewEntity(entityName, entityType, entityName)
```

Comment:
we will need the host:port of the target somewhere here , not sure if we could add it after